koa-crud/test/integration/data-access/vendor.spec.ts

```diff
@@ -0,0 +1,360 @@
+    afterEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    beforeEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    it('should return a list of vendors', async () => {
+      const mock = {
+        name: 'Luis Angelo Belmonte',
+        type: VendorType.Transfer,
+        dateTimeCreated: new Date(),
+        dateTimeUpdated: new Date(),
+      };
+      await insertOneVendor(mock);
+
+      await expect(selectAllVendors()).eventually.fulfilled.and.have.length(1);
+    });
+  });
+
+  describe('Select One Vendor By Filters', () => {
+    afterEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    beforeEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    describe('GIVEN existent filters', () => {
+      it('should return the vendor', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        const main = await insertOneVendor(mock);
+        await expect(selectOneVendorByFilters({ _id: main._id }))
+          .to.eventually.fulfilled.and.property('_id')
+          .eqls(main._id);
+      });
+    });
+
+    describe('GIVEN non existent filters', () => {
+      it('should return the vendor', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await insertOneVendor(mock);
+        await expect(selectOneVendorByFilters({ _id: mockedId })).to.eventually
+          .fulfilled.and.null;
+      });
+    });
+
+    describe('GIVEN invalid filters', () => {
+      it('should return the vendor', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await insertOneVendor(mock);
+        await expect(selectOneVendorByFilters({ _id: 'qwe' })).to.eventually
+          .rejected;
+      });
+    });
+  });
+
+  describe('Updating a vendor', () => {
+    describe('GIVEN correct inputs and SEAMLESS type', () => {
+      it('should update the vendor to SEAMLESS', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        const main = await insertOneVendor(mock);
+
+        await expect(
+          updateVendorByFilters(
+            { _id: main._id },
+            {
+              name: 'Luis Angelo Belmonte',
+              type: VendorType.Seamless,
+              dateTimeCreated: new Date(),
+              dateTimeUpdated: new Date(),
+            },
+          ),
+        )
+          .to.eventually.fulfilled.property('type')
+          .eqls('SEAMLESS');
```

Comment:
You can shorten this to `property('type', 'SEAMLESS')`

Reply:
Noted on this sir.